platform_channels/ios/Runner/AppDelegate.swift

```diff
@@ -45,7 +45,49 @@ import Flutter
         
         FlutterEventChannel(name: "eventChannelDemo", binaryMessenger: flutterViewController.binaryMessenger).setStreamHandler(AccelerometerStreamHandler())
         
+        var petList : [[String:String]] = []
+        
+        // A FlutterBasicMessageChannel for sending petList to Dart.
+        let stringCodecChannel = FlutterBasicMessageChannel(name: "stringCodecDemo", binaryMessenger: flutterViewController.binaryMessenger, codec: FlutterStringCodec.sharedInstance())
+        
+        // Registers a MessageHandler for FlutterBasicMessageChannel to receive pet details to be added in petList and send the it back to Dart using
+        // stringCodecChannel.
+        FlutterBasicMessageChannel(name: "jsonMessageCodecDemo", binaryMessenger: flutterViewController.binaryMessenger, codec: FlutterJSONMessageCodec.sharedInstance())
+            .setMessageHandler{(message: Any?, reply: FlutterReply) -> Void in
+                petList.insert(message! as! [String: String], at: 0)
+                stringCodecChannel.sendMessage(self.convertPetListToJson(petList: petList))
+            }
+        
+        
+        // Registers a MessageHandler for FlutterBasicMessageChannel to receive the index of pet details to be removed from the petList and send the petList back to Dart using
```

Comment:
nit: "Registers a MessageHandler for FlutterBasicMessageHandler to receive indices of detail records to remove from the petList."

If you'd like to keep the bits about sending the list back and returning nile, I'd move those comments to be just above lines 75 and 77, respectively.

Reply:
I'll remove the extra information. The documentation on line 50 cover that :D